libcalico-go/lib/clientv3/client.go

```diff
@@ -393,43 +395,29 @@ func (c client) ensureClusterInformation(ctx context.Context, calicoVersion, clu
 	return nil
 }
 
-// ensureDefaultTierExists ensures that the "default" Tier exits in the datastore.
-// This is done by trying to create the default tier. If it doesn't exists, it
-// is created.  A error is returned if there is any error other than when the
-// default tier resource already exists.
-func (c client) ensureDefaultTierExists(ctx context.Context) error {
-	order := v3.DefaultTierOrder
-	defaultTier := v3.NewTier()
-	defaultTier.ObjectMeta = metav1.ObjectMeta{Name: names.DefaultTierName}
-	defaultTier.Spec = v3.TierSpec{
-		Order: &order,
-	}
-	if _, err := c.Tiers().Create(ctx, defaultTier, options.SetOptions{}); err != nil {
-		if _, ok := err.(cerrors.ErrorResourceAlreadyExists); !ok {
-			return err
-		}
-	}
-	return nil
-}
-
-// ensureAdminNetworkPolicyTierExists ensures that the "adminnetworkpolicy" Tier exits in the datastore.
-// This is done by trying to create the adminnetworkpolicy tier. If it doesn't exists, it
-// is created.  A error is returned if there is any error other than when the
+// ensureTierExists ensures that a desired Tier exits in the datastore.
+// This is done by first trying to get the tier. If it doesn't exist, it
+// is created. A error is returned if there is any error other than when the
 // tier resource already exists.
-func (c client) ensureAdminNetworkPolicyTierExists(ctx context.Context) error {
-	order := v3.AdminNetworkPolicyTierOrder
-	actionPass := v3.Pass
-	anpTier := v3.NewTier()
-	anpTier.ObjectMeta = metav1.ObjectMeta{Name: names.AdminNetworkPolicyTierName}
-	anpTier.Spec = v3.TierSpec{
+func (c client) ensureTierExists(ctx context.Context, name string, defaultAction v3.Action, order float64) error {
+	tier, err := c.Tiers().Get(ctx, name, options.GetOptions{})
+	if err == nil && tier != nil {
```

Comment:
What if err != nil? We don't seem to handle that case - I think we need to check specifically for `IsNotFound` errors, and then log out any other types of errors.

Reply:
Removed it, since it feels duplicate since we check tier existence few lines after.